apis/v1alpha1/resolver_endpoint.go

```diff
@@ -0,0 +1,163 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ResolverEndpointSpec defines the desired state of ResolverEndpoint.
+//
+// In the response to a CreateResolverEndpoint (https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_CreateResolverEndpoint.html),
+// DeleteResolverEndpoint (https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_DeleteResolverEndpoint.html),
+// GetResolverEndpoint (https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_GetResolverEndpoint.html),
+// Updates the name, or ResolverEndpointType for an endpoint, or UpdateResolverEndpoint
+// (https://docs.aws.amazon.com/Route53/latest/APIReference/API_route53resolver_UpdateResolverEndpoint.html)
+// request, a complex type that contains settings for an existing inbound or
+// outbound Resolver endpoint.
+type ResolverEndpointSpec struct {
+
+	// A unique string that identifies the request and that allows failed requests
+	// to be retried without the risk of running the operation twice. CreatorRequestId
+	// can be any unique string, for example, a date/time stamp.
+	// +kubebuilder:validation:Required
+	CreatorRequestID *string `json:"creatorRequestID"`
```

Comment:
Need to ignore this field from the spec - it's marked as required (so the api-server rejects your testing resource) while it's not (you're setting this with a custom ID, in hooks.go)

Reply:
Done.